pymatgen/core/structure.py

```diff
@@ -3980,68 +3986,67 @@ def set_charge(self, new_charge: float = 0.0) -> None:
 
     def relax(
         self,
-        calculator: str = "m3gnet",
+        calculator: str | Calculator = "m3gnet",
```

Comment:
I think we should add CHGNet as another `str` option here and even make it the default considering it's easier to install on account of not needing TF (PyTorch causes me less trouble anyways).

Reply:
Happy to add support for CHGNet. Would have to get @shyuep's approval to change the default, as that'd be a breaking change (and I imagine he has personal preference about M3GNet).

Reply:
Yes on supporting CHGNet. No on changing default. In any case, we will replace the m3gnet calculator with the Matgl version soon. That will be Pytorch.

Reply:
Actually, it seems like CHGNet by default takes a `Structure` object in as input. Obviously that's not a problem, but it'd require some modification of the logic here which converts between ASE `Atoms` and `Structure`. I think that's best left to a future PR.

Reply:
> Actually, it seems like CHGNet by default takes a Structure object in as input.

Ah yes, forget to mention that since I already started implementing and noticed it would take some reworking.

> I think that's best left to a future PR.

Agreed.